### DISCUSSION TITLE
Retrieving Multi Properties.

### DISCUSSION THREAD

**Discussion by navilan** [Gremlinq/ExRam.Gremlinq#123] [Q&A] [answered]
Hi, 

Thank you for the fantastic library . This helped me get going really fast. Much appreciated (and Sponsored!). I'd love to see some examples on how to retrieve multi properties. Right now when I use a projection and use `.By(x => x.arrayValues)`, I get only the first item in the array.  

Cheers
Navilan (Puthir)

**Comment by danielcweber** [comment, marked as answer]
Hi,

thanks for the sponsoring.

As far as I can see it, every operator that does projections with 'By' will only ever consider the first value on the traversal. If you need to project to all property values, try aggregating (collecting, folding, you name it), like ....By(__ => __.Values(x => x.SomeProperty).Fold()).

More info on the Gremlin fold-step: https://tinkerpop.apache.org/docs/current/reference/#fold-step

**Reply by navilan** [reply]
@danielcweber  - That worked great. Thank you.
